addons/misc/functions/fnc_unpackFAK.sqf

```diff
@@ -0,0 +1,6979 @@
+            };
+            case 8: { 
+                _unit removeItem _mag;
+                [_unit, "kat_MFAK_Magazine", "", Slot_1_2_3_4_5_6_7] call ACEFUNC(common,addToInventory);
+                [_unit, _eighthSlotItemMFAK, _container] call FUNC(arrayToInvItem);
+            };
+        };
+    };
+};
+
+// ----- Handeling for magazine -----
+
+private _ammoCount = [_unit, _mag, true] call FUNC(getMagazineAmmoCounts);
+private _removeFromArray = [];
+private _firstSlotItem_1 = "";
+private _secondSlotItem_2 = "";
+private _thirdSlotItem_3 = "";
+private _fourthSlotItem_4 = "";
+private _fifthSlotItem_5 = "";
+private _sixthSlotItem_6 = "";
+private _seventhSlotIte_7 = "";
+private _eighthSlotItem_8 = "";
+
+switch (_mag) do {
+    case "kat_IFAK_Magazine": { 
+        _firstSlotItem_1 = _firstSlotItemIFAK;
+        _secondSlotItem_2 = _secondSlotItemIFAK;
+        _thirdSlotItem_3 = _thirdSlotItemIFAK;
+        _fourthSlotItem_4 = _fourthSlotItemIFAK;
+
+        switch (_containerIFAK) do {
+            case 0: { _container = "";};
+            case 1: { _container = "uniform";};
+            case 2: { _container = "vest";};
+            case 3: { _container = "backpack";};
+        };
+    };
+
+    case "kat_AFAK_Magazine": { 
+        _firstSlotItem_1 = _firstSlotItemAFAK;
+        _secondSlotItem_2 = _secondSlotItemAFAK;
+        _thirdSlotItem_3 = _thirdSlotItemAFAK;
+        _fourthSlotItem_4 = _fourthSlotItemAFAK;
+        _fifthSlotItem_5 = _fifthSlotItemAFAK;
+        _sixthSlotItem_6 = _sixthSlotItemAFAK;
+
+        switch (_containerAFAK) do {
+            case 0: { _container = "";};
+            case 1: { _container = "uniform";};
+            case 2: { _container = "vest";};
+            case 3: { _container = "backpack";};
+        };
+    };
+
+    case "kat_MFAK_Magazine": { 
+        _firstSlotItem_1 = _firstSlotItemMFAK;
+        _secondSlotItem_2 = _secondSlotItemMFAK;
+        _thirdSlotItem_3 = _thirdSlotItemMFAK;
+        _fourthSlotItem_4 = _fourthSlotItemMFAK;
+        _fifthSlotItem_5 = _fifthSlotItemMFAK;
+        _sixthSlotItem_6 = _sixthSlotItemMFAK;
+        _seventhSlotIte_7 = _seventhSlotItemMFAK;
+        _eighthSlotItem_8 = _eighthSlotItemMFAK;
+
+        switch (_containerMFAK) do {
+            case 0: { _container = "";};
+            case 1: { _container = "uniform";};
+            case 2: { _container = "vest";};
+            case 3: { _container = "backpack";};
+        };
+    };
+};
+
+// Deletes the false ammo values out of ammo count 
+
+if ((count _ammoCount) > 1) then {
+    {
+        if !([_unit, _mag, _slotsToGive, (_x select 0)] call FUNC(checkFAKSlot)) then {
+            _removeFromArray pushBack _x;
+        };
+    } forEach _ammoCount;
+
+    _ammoCount = _ammoCount - _removeFromArray;
+    _ammoCount sort true;
+};
+
+_ammoCount = _ammoCount select 0;
+_removeContainer = _ammoCount select 1;
+_ammoCount = _ammoCount select 0;
+
+switch (_slotsToGive) do {
+    
+    case 0: { 
```

Comment:
please add some comments on what case state is for what

Reply:
Pretty self explanatory if you ask me.
Slot 0 =  unpack the rest of all slots left. ( I give you that one, this one was hard to guess)
Slot 1= unpack slot 1
Slot 2 = unpack slot 2
-||- (and so on...)